app/src/main/java/org/bepass/oblivion/ConnectionState.java

```diff
@@ -1,5 +1,9 @@
 package org.bepass.oblivion;
 
 public enum ConnectionState {
-    CONNECTING, CONNECTED, DISCONNECTED
+    CONNECTING, CONNECTED, DISCONNECTED;
+
+    public boolean isConnectedOrConnecting() {
+        return this == CONNECTED || this == CONNECTING;
+    }
 }
```

Comment:
If we instead made a `isDisconnected()` boolean method, I think the rest of the code would be simpler.